cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/CachedPrecompileService.java

```diff
@@ -0,0 +1,43 @@
+package io.cloudslang.lang.compiler;
+
+import io.cloudslang.lang.compiler.modeller.result.ExecutableModellingResult;
+import org.springframework.stereotype.Component;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import javax.annotation.PostConstruct;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by bancl on 10/10/2016.
+ */
+@Component
+public class CachedPrecompileService {
+
+    private Cache<String, ExecutableModellingResult> cache;
+
+    @PostConstruct
+    public void init() {
+        cache = CacheBuilder.newBuilder()
+                .maximumSize(500)
+                .concurrencyLevel(10)
```

Comment:
maybe the concurrencyLevel should be `2 * Runtime.getRuntime().availableProcessors()`. WDYT?